### PR TITLE
Last nits before shipping 🛳️ 

### PR DESCRIPTION
**Pull request by merveenoyan**

### CODE REVIEW

**Comment by snehilsanyal** [comment]
Hey @merveenoyan I went through my chapters, can you fix these things (or let me know if I should open a PR separately):

**Ethics and Bias Unit**
- In the [introduction](https://github.com/johko/computer-vision-course/blob/main/chapters/en/Unit%2012%20-%20Ethics%20and%20Biases/introduction.mdx) chapter, there are some spacing issues [here](https://github.com/johko/computer-vision-course/blob/9a5c98370015468b81bbabaa089bd0a96f1f5d03/chapters/en/Unit%2012%20-%20Ethics%20and%20Biases/introduction.mdx#L57), [here](https://github.com/johko/computer-vision-course/blob/9a5c98370015468b81bbabaa089bd0a96f1f5d03/chapters/en/Unit%2012%20-%20Ethics%20and%20Biases/introduction.mdx#L65) and [here](https://github.com/johko/computer-vision-course/blob/9a5c98370015468b81bbabaa089bd0a96f1f5d03/chapters/en/Unit%2012%20-%20Ethics%20and%20Biases/introduction.mdx#L70) this is due to missing new lines between the bullets.
image
- In the [HF Efforts]() chapter, there is a spacing issue [here](https://github.com/johko/computer-vision-course/blob/9a5c98370015468b81bbabaa089bd0a96f1f5d03/chapters/en/Unit%2012%20-%20Ethics%20and%20Biases/conclusion.mdx#L7) due to missing new lines as well.
image

**Multimodal Models Unit**
In the [introduction](https://github.com/johko/computer-vision-course/blob/main/chapters/en/Unit%204%20-%20Multimodal%20Models/introduction.mdx#L28) chapter, only one issue with one of the [images](https://huggingface.co/datasets/hf-vision/course-assets/blob/main/multimodal_fusion_text_vision/bigbang.jpg) which is not rendering [here](https://github.com/johko/computer-vision-course/blob/9a5c98370015468b81bbabaa089bd0a96f1f5d03/chapters/en/Unit%204%20-%20Multimodal%20Models/introduction.mdx#L28)

**Comment by charchit7** [comment]
In Unit 5 : Generative Models, Section on VAE : https://github.com/johko/computer-vision-course/pull/131 there's an issue with the latex equation. @sarthak247 could you please have a look :)

**Comment by charchit7** [comment]
In Unit 5 : Introduction to CycleGAN #158 The link reference to DCGAN paper bracket is not closed. Let me know if I can raise a PR.

**Comment by merveenoyan** [comment]
@snehilsanyal @charchit7 feel free to open PRs 😊

File: chapters/en/Unit 1 - Fundamentals/feature-extraction/feature-matching.mdx [hunk 1]
```diff
@@ -179,9 +150,7 @@ kp1, des1 = sift.detectAndCompute(img1, None)
 kp2, des2 = sift.detectAndCompute(img2, None)
 ```
 
-Define the FLANN parameters
-
-- Trees is the numbers of bins you want.
+We will now define the FLANN parameters. Here, trees is the numbers of bins you want.
```
Comment (MKhalusova) [inline chapters/en/Unit 1 - Fundamentals/feature-extraction/feature-matching.mdx:153]:
```suggestion
We will now define the FLANN parameters. Here, trees is the number of bins you want.
```

File: chapters/en/Unit 1 - Fundamentals/image_and_imaging/imaging.mdx [hunk 2]
```diff
@@ -23,19 +23,25 @@ The resolution and quality of a digital image significantly depend on the follow
...
+
 
 ## Image Restoration and Reconstruction Techniques
 Image restoration focuses on recovering a degraded image using knowledge about the degradation phenomenon. It often involves modelling the degradation process and applying inverse processes to regain the original image.
 
-![An example for image restoration where the image is restored and colorized](https://huggingface.co/datasets/hf-vision/course-assets/resolve/4def8c412ee6b08f4522e818a0474d155363d87b/pic_8.jpg)
-
+<div class="flex justify-center">
+    <img src="https://huggingface.co/datasets/hf-vision/course-assets/resolve/4def8c412ee6b08f4522e818a0474d155363d87b/pic_8.jpg" alt="An example for image restoration where the image is restored and colorized">
+</div>
```
Comment (MKhalusova) [inline chapters/en/Unit 1 - Fundamentals/image_and_imaging/imaging.mdx:44]:
```suggestion
</div>

```

File: chapters/en/Unit 12 - Ethics and Biases/ethics-bias-ai.mdx [hunk 3]
```diff
@@ -29,7 +29,7 @@ From the previous example, a few aspects of this technology to keep in mind woul
 - Awareness about these algorithms and their after-effects.
 
 <Tip>
-💡Check out The Consentful Tech Project [here](https://www.consentfultech.io/). This project raises awareness, develops strategies, and shares skills to help people build and use technology consentfully.
+💡Check out The Consentful Tech Project <a href="https://www.consentfultech.io/">here</a>. This project raises awareness, develops strategies, and shares skills to help people build and use technology consentfully.
```
Comment (MKhalusova) [inline chapters/en/Unit 12 - Ethics and Biases/ethics-bias-ai.mdx:32]:
```suggestion

💡Check out The Consentful Tech Project [here](https://www.consentfultech.io/). This project raises awareness, develops strategies, and shares skills to help people build and use technology consentfully.
```

File: chapters/en/Unit 12 - Ethics and Biases/ethics-bias-ai.mdx [hunk 4]
```diff
@@ -57,7 +57,7 @@ Asimov's laws of robotics were one of the earliest works in ethics for technolog
 23 Asilomar AI Principles for Responsible AI Development
 
 <Tip>
-💡You can check the full list of the 23 Asilomar AI Principles and the signatories [here](https://futureoflife.org/open-letter/ai-principles/).
+💡You can check the full list of the 23 Asilomar AI Principles and the signatories <a href="https://futureoflife.org/open-letter/ai-principles/)">here</a>.
```
Comment (MKhalusova) [inline chapters/en/Unit 12 - Ethics and Biases/ethics-bias-ai.mdx:60]:
```suggestion

💡You can check the full list of the 23 Asilomar AI Principles and the signatories [here](https://futureoflife.org/open-letter/ai-principles/).
```

File: chapters/en/Unit 12 - Ethics and Biases/ethics-bias-ai.mdx [hunk 5]
```diff
@@ -86,7 +86,7 @@ The ten core principles that lay out a human-rights centered approach to Ethics
 - **Fairness and Non-Discrimination:** AI actors should promote social justice, fairness, and non-discrimination while taking an inclusive approach to ensure AI's benefits are accessible to all.
 
 <Tip>
-💡To read the complete report by UNESCO on "Recommendations on the Ethics of Artificial Intelligence", you can visit [here](https://unesdoc.unesco.org/ark:/48223/pf0000381137).
+💡To read the complete report by UNESCO on "Recommendations on the Ethics of Artificial Intelligence", you can visit <a href="https://unesdoc.unesco.org/ark:/48223/pf0000381137">here</a>.
```
Comment (MKhalusova) [inline chapters/en/Unit 12 - Ethics and Biases/ethics-bias-ai.mdx:89]:
```suggestion

💡To read the complete report by UNESCO on "Recommendations on the Ethics of Artificial Intelligence", you can visit [here](https://unesdoc.unesco.org/ark:/48223/pf0000381137).
```

File: chapters/en/Unit 12 - Ethics and Biases/introduction.mdx [hunk 6]
```diff
@@ -23,7 +23,7 @@ ImageNet is a large-scale dataset that was created for object recognition benchm
...
 
 <Tip>
-💡You can read more about ImageNet [here](https://www.image-net.org/). Also check out [this TED Talk](https://www.youtube.com/watch?v=40riCqvRoMs) by Prof. Fei Fei Li on the same topic.
+💡You can read more about ImageNet <a href="https://www.image-net.org/">here</a>. Also check out <a href="https://www.youtube.com/watch?v=40riCqvRoMs">this TED talk</a> by Prof. Fei Fei Li on the same topic.
```
Comment (MKhalusova) [inline chapters/en/Unit 12 - Ethics and Biases/introduction.mdx:26]:
```suggestion

💡You can read more about ImageNet [here](https://www.image-net.org/). Also check out [this TED Talk](https://www.youtube.com/watch?v=40riCqvRoMs) by Prof. Fei Fei Li on the same topic.
```

File: chapters/en/Unit 12 - Ethics and Biases/ethics-bias-ai.mdx (hunk 3 above)
Comment (MKhalusova) [reply chapters/en/Unit 12 - Ethics and Biases/ethics-bias-ai.mdx:32]: The link should render properly if there's a new line after the <Tip> tag

File: chapters/en/Unit 12 - Ethics and Biases/ethics-bias-ai.mdx (hunk 4 above)
Comment (MKhalusova) [reply chapters/en/Unit 12 - Ethics and Biases/ethics-bias-ai.mdx:60]: New line should help

File: chapters/en/Unit 12 - Ethics and Biases/ethics-bias-ai.mdx (hunk 5 above)
Comment (MKhalusova) [reply chapters/en/Unit 12 - Ethics and Biases/ethics-bias-ai.mdx:89]: New line after the <Tip> should help with link rendering.

File: chapters/en/Unit 12 - Ethics and Biases/introduction.mdx [hunk 7]
```diff
@@ -81,7 +81,7 @@ b. Image obfuscation techniques like blurring and mosaicing were applied to thes
 c. It was shown that these images lead to a very minimal drop of accuracy while benchmarking on object recognition tasks and are suitable for training privacy-aware visual classifiers.
 
 <Tip>
-💡For more details on the ImageNet Roulette experiment, you can follow the article on ImageNet Roulette. The experiment is posted on [Excavating AI](https://excavating.ai/) which discusses this in detail. To know more about ImageNet's stance and research on mitigating these issues, you can take a look into the full technical report submitted by them [here](https://www.image-net.org/filtering-and-balancing/).
+💡For more details on the ImageNet Roulette experiment, you can follow the article on ImageNet Roulette. The experiment is posted on <a href="https://excavating.ai/">Excavating AI</a> which discusses this in detail. To know more about ImageNet's stance and research on mitigating these issues, you can take a look into the full technical report submitted by them [here](https://www.image-net.org/filtering-and-balancing/).
```
Comment (MKhalusova) [inline chapters/en/Unit 12 - Ethics and Biases/introduction.mdx:84]:
```suggestion

💡For more details on the ImageNet Roulette experiment, you can follow the article on ImageNet Roulette. The experiment is posted on [Excavating AI](https://excavating.ai/) which discusses this in detail. To know more about ImageNet's stance and research on mitigating these issues, you can take a look into the full technical report submitted by them [here](https://www.image-net.org/filtering-and-balancing/).
```

File: chapters/en/Unit 13 - Outlook/hyena.mdx [hunk 8]
```diff
@@ -69,7 +69,7 @@ This network takes the positional index and potentially positional encodings as
 This implies that instead of learning the values of the convolution filter directly, we learn a mapping from a temporal positional encoding to the values, which is more computationally efficient, especially for long sequences.
 
 <Tip>
-It's important to note that the mapping function can be conceptualized within various abstract models, such Neural Field or State Space Models (S4) as discussed in [H3 paper](https://arxiv.org/abs/2212.14052).
+It's important to note that the mapping function can be conceptualized within various abstract models, such Neural Field or State Space Models (S4) as discussed in <a href="https://arxiv.org/abs/2212.14052">H3 Paper</a>.
```
Comment (MKhalusova) [inline chapters/en/Unit 13 - Outlook/hyena.mdx:72]:
```suggestion

It's important to note that the mapping function can be conceptualized within various abstract models, such Neural Field or State Space Models (S4) as discussed in [H3 paper](https://arxiv.org/abs/2212.14052).
```

File: chapters/en/Unit 3 - Vision Transformers/MobileVIT.mdx [hunk 9]
```diff
@@ -57,9 +57,9 @@ This is a massive improvement!
 <Tip>
```
Comment (MKhalusova) [inline chapters/en/Unit 3 - Vision Transformers/MobileVIT.mdx:57]:
```suggestion
<Tip>

```

File: chapters/en/Unit 3 - Vision Transformers/MobileVIT.mdx [hunk 10]
```diff
@@ -57,9 +57,9 @@ This is a massive improvement!
...
 There have been many other forms of Attention such that complexity has ranged from O(k), O(k*sqrt(k)), O(k*log(k)).
 
-Separable self-attention was also not the first paper to have O(k) complexity. In Linformer, O(k) complexity for Attention was also achived in [Linformer](https://arxiv.org/abs/2006.04768) before separable self-attention.
+Separable self-attention was also not the first paper to have O(k) complexity. In Linformer, O(k) complexity for Attention was also achived in <a href="https://arxiv.org/abs/2006.04768">Linformer</a> before separable self-attention.
```
Comment (MKhalusova) [inline chapters/en/Unit 3 - Vision Transformers/MobileVIT.mdx:60]:
```suggestion
Separable self-attention was also not the first paper to have O(k) complexity. In Linformer, O(k) complexity for Attention was also achived in [Linformer](https://arxiv.org/abs/2006.04768) before separable self-attention.
```

Comment (MKhalusova) [reply chapters/en/Unit 3 - Vision Transformers/MobileVIT.mdx:60]: I added the newline after the <Tip> in a separate suggestion.

File: chapters/en/Unit 3 - Vision Transformers/Swin Transformer.mdx [hunk 11]
```diff
@@ -3,7 +3,7 @@ Introduced in the 2021 paper, [Swin Transformer: Hierarchical Vision Transformer
 
 <Tip>
 
-A backbone, in terms of deep learning, is a part of a neural network that does feature extraction. Additional layers can be added to the backbone to do a variety of vision tasks. Hierarchical backbones have tiered structures, sometimes with varying resolutions. This is in contrast to the non-hierarchical **plain backbone** in [VitDet](https://arxiv.org/abs/2203.16527) model.
+A backbone, in terms of deep learning, is a part of a neural network that does feature extraction. Additional layers can be added to the backbone to do a variety of vision tasks. Hierarchical backbones have tiered structures, sometimes with varying resolutions. This is in contrast to the non-hierarchical **plain backbone** in <a href="https://arxiv.org/abs/2203.16527">VitDet</a> model.
```
Comment (MKhalusova) [inline chapters/en/Unit 3 - Vision Transformers/Swin Transformer.mdx:6]:
```suggestion
A backbone, in terms of deep learning, is a part of a neural network that does feature extraction. Additional layers can be added to the backbone to do a variety of vision tasks. Hierarchical backbones have tiered structures, sometimes with varying resolutions. This is in contrast to the non-hierarchical **plain backbone** in [VitDet](https://arxiv.org/abs/2203.16527) model.
```

File: chapters/en/Unit 4 - Multimodal Models/introduction.mdx [hunk 12]
```diff
@@ -68,7 +68,7 @@ Hugging Face supports a wide variety of multimodal tasks. Let us look into some
 
 <Tip>
```
Comment (MKhalusova) [inline chapters/en/Unit 4 - Multimodal Models/introduction.mdx:69]:
```suggestion
<Tip>

```

File: chapters/en/Unit 4 - Multimodal Models/introduction.mdx [hunk 13]
```diff
@@ -68,7 +68,7 @@ Hugging Face supports a wide variety of multimodal tasks. Let us look into some
...
 💡An amazing usecase of multimodal task is Multimodal Emotion Recognition (MER). The MER task involves recognition of emotion from two or more modalities like audio+text, text+vision, audio+vision or vision+text+audio As we discussed in the example, MER is more efficient than unimodal emotion recognition and gives clear insight into the 
-emotion recognition task. Check out more on MER with this repository: https://github.com/EvelynFan/AWESOME-MER. 
+emotion recognition task. Check out more on MER with <a href="https://github.com/EvelynFan/AWESOME-MER">this repository</a>.
```
Comment (MKhalusova) [inline chapters/en/Unit 4 - Multimodal Models/introduction.mdx:71]:
```suggestion
emotion recognition task. Check out more on MER with [this repository]( https://github.com/EvelynFan/AWESOME-MER).
```

File: chapters/en/Unit 4 - Multimodal Models/introduction.mdx [hunk 14]
```diff
@@ -87,7 +87,7 @@ Vision Language Models (VLMs) are models that can understand and process both vi
 
 
 <Tip>
-💡Meta released first multimodal AI model to bind information from 6 different modalities:  images and videos, audio, text, depth, thermal, and inertial measurement units (IMUs). Check more here: https://imagebind.metademolab.com/
+💡Meta released first multimodal AI model to bind information from 6 different modalities: images and videos, audio, text, depth, thermal, and inertial measurement units (IMUs). Learn more about it <a href="https://imagebind.metademolab.com/" target="_blank">here</a>.
```
Comment (MKhalusova) [inline chapters/en/Unit 4 - Multimodal Models/introduction.mdx:90]:
```suggestion

💡Meta released first multimodal AI model to bind information from 6 different modalities: images and videos, audio, text, depth, thermal, and inertial measurement units (IMUs). Learn more about it [here](https://imagebind.metademolab.com/).
```

**Review by MKhalusova — COMMENTED** [review]
Thanks for the fixes, I left some comments.

**Comment by hanouticelina** [comment]
Hello @merveenoyan , 
I went through [Unit 3- Vision Transformers/Vision Transformers for Image Segmentation.mdx](https://github.com/johko/computer-vision-course/blob/f748bc6f39b949ce5953175587de8bbb4592dbb2/chapters/en/Unit%203%20-%20Vision%20Transformers/Vision%20Transformers%20for%20Image%20Segmentation.mdx). I noticed _minor_ nits to fix :

- Line 16 : which Transformers aims to solve -> which Transformers **aim** to solve
- Line 93 : these techniques -> **These** techniques (beginning of a new sentence)


Thank you!

**Comment by lulmer** [comment]
Hey @merveenoyan 
For Unit 13 : Hyena, there are problems with the latex symbol spacing (almost all latex blocks are affected) and greek letters not rendering properly (in the screenshot "gamma" and "theta" should be rendered in greek).

image

**Comment by merveenoyan** [comment]
@lulmer problem is with docbuilder and not our side

**Comment by merveenoyan** [comment]
@hanouticelina can you open a PR to this branch?

**Comment by merveenoyan** [comment]
@charchit7 @snehilsanyal can you open a PR to this branch or main branch please?

**Comment by vasugupta9** [comment]
Hi @merveenoyan , 

There are few minor rendering issues related to equations like in section 'Simplified Solution' , etc - [here](https://moon-ci-docs.huggingface.co/docs/computer-vision-course/pr_228/en/Unit%208%20-%203D%20Vision,%20Scene%20Rendering%20and%20Reconstruction/3d_measurements_stereo_vision )

I would like to correct these but last time I tried I had issues getting doc builder to run properly on my system. 

Is there an easy way where I can just upload/point to a single markdown file and see its rendering without having to worry about correctness of toc-tree, etc ? 

If there isn't such a tool, do you think it would be good to have one ? If so, I'd be happy to spend some time understanding the existing tool/flow and build this additional feature.

Thanks !

**Comment by hanouticelina** [comment]
> @hanouticelina can you open a PR to this branch?

Hello @merveenoyan,
the PR is open : https://github.com/johko/computer-vision-course/pull/240

**Comment by merveenoyan** [comment]
@vasugupta9 you don't need to run doc-builder on your local, just modify the number of PR here https://moon-ci-docs.huggingface.co/docs/computer-vision-course/pr_224/en/Unit%200%20-%20Welcome/welcome

**Comment by vasugupta9** [comment]
> @vasugupta9 you don't need to run doc-builder on your local, just modify the number of PR here (your latest PR is hosted here https://moon-ci-docs.huggingface.co/docs/computer-vision-course/pr_241/en/Unit%200%20-%20Welcome/welcome)

yes, this helped a lot with fixing the rendering.

**Review by MKhalusova — APPROVED** [review]

**Review by ATaylorAerospace — APPROVED** [review]
Updates Approved

**Review by ratan — COMMENTED** [review]
Looks good

**Pull request merged**